packages/experimental/src/rest/BaseResource.ts

```diff
@@ -0,0 +1,248 @@
+
+  /** Get the url for a SimpleResource
+   *
+   * Default implementation conforms to common REST patterns
+   */
+  static url(urlParams: Readonly<Record<string, any>>): string {
+    if (
+      Object.prototype.hasOwnProperty.call(urlParams, 'url') &&
+      urlParams.url &&
+      typeof urlParams.url === 'string'
+    ) {
+      return urlParams.url;
+    }
+    if (this.pk(urlParams as any) !== undefined) {
+      return `${this.urlRoot.replace(/\/$/, '')}/${this.pk(urlParams as any)}`;
+    }
+    return this.urlRoot;
+  }
+
+  /** Get the url for many SimpleResources
+   *
+   * Default implementation conforms to common REST patterns
+   */
+  static listUrl(
+    searchParams: Readonly<Record<string, string | number | boolean>> = {},
+  ): string {
+    if (Object.keys(searchParams).length) {
+      return `${this.urlRoot}?${paramsToString(searchParams)}`;
+    }
+    return this.urlRoot;
+  }
+
+  /** Perform network request and resolve with HTTP Response */
+  static fetchResponse(input: RequestInfo, init: RequestInit) {
+    let options: RequestInit = init;
+    if (!options.body || typeof options.body === 'string') {
+      options = {
+        ...options,
+        headers: {
+          'Content-Type': 'application/json',
+          ...options.headers,
+        },
+      };
+    }
+    return fetch(input, options)
+      .then(response => {
+        if (!response.ok) {
+          throw new NetworkError(response);
+        }
+        return response;
+      })
+      .catch(error => {
+        // ensure CORS, network down, and parse errors are still caught by NetworkErrorBoundary
+        if (error instanceof TypeError) {
+          (error as any).status = 400;
+        }
+        throw error;
+      });
+  }
+
+  /** Perform network request and resolve with json body */
+  static fetch(input: RequestInfo, init: RequestInit) {
+    return this.fetchResponse(input, init).then((response: Response) => {
+      if (
+        !response.headers.get('content-type')?.includes('json') ||
+        response.status === 204
+      )
+        return response.text();
+      return response.json().catch(error => {
+        error.status = 400;
+        throw error;
+      });
+    });
+  }
+
+  /** Init options for fetch - run at fetch */
+  static getFetchInit(init: Readonly<RequestInit>): RequestInit {
+    return init;
+  }
+
+  /** Init options for fetch - run at render */
+  static useFetchInit(init: Readonly<RequestInit>): RequestInit {
+    return init;
+  }
+
+  /** Get the request options for this SimpleResource */
+  static getEndpointExtra(): EndpointExtraOptions | undefined {
+    return;
+  }
+
+  /** Field where endpoint cache is stored */
+  protected static readonly cacheSymbol = Symbol('memo');
+
+  /** Used to memoize endpoint methods
+   *
+   * Relies on existance of runInit() member.
+   */
+  protected static memo<T>(name: string, construct: () => T): T {
+    if (!Object.hasOwnProperty.call(this, this.cacheSymbol)) {
```

Comment:
```suggestion
    if (!Object.prototype.hasOwnProperty.call(this, this.cacheSymbol)) {
```
(altho this will work because Object inherits from Function.prototype which inherits from Object.prototype)

Reply:
this is copy/paste from @rest-hooks/rest. i'll keep this in mind for future updates tho.